nautobot/utilities/tables.py

```diff
@@ -58,6 +60,12 @@ def __init__(self, *args, user=None, **kwargs):
                 )
             # symmetric relationships are already handled above in the source_type case
 
+        if model := getattr(self.Meta, "model", None):
+            # Disable ordering on these models(Tree Models) as sorting on UI results in rows appears
+            # to be children of the wrong parent rows.
+            if issubclass(model, (MPTTModel, TreeNode)):
```

Comment:
This would remove functionality for `Region` which works* today, as `MPTTModel` supports it.

This would make it more obvious for `Location` and other `TreeNode` based models that they do no support it, because under the hood you can't perform `order_by`: 
- https://github.com/matthiask/django-tree-queries/issues/23
- https://github.com/matthiask/django-tree-queries/issues/33


*Subjective to what any person would expect sorting to behave like here with the way we return nested results.